App/stores/location.tsx

```diff
@@ -101,12 +110,25 @@ export function LocationContextProvider({
 						)}`
 					);
 					setGpsLocation(location);
-					setCurrentLocation(location);
+					setAndSaveCurrentLocation(location);
 
 					return T.of(undefined);
 				}
 			)
-		)().catch(sentryError('LocationContextProvider'));
+		)().catch(() => {
+			async function useLastKnownLocation(): Promise<void> {
+				const location_string = await AsyncStorage.getItem(
+					'LAST_KNOWN_LOCATION'
+				);
+
+				if (location_string !== null) {
+					setCurrentLocation(JSON.parse(location_string));
+				}
+			}
+			useLastKnownLocation().catch(
+				sentryError('LocationContextProvider')
+			);
+		});
```

Comment:
I'm trying to use fp-ts functional programming paradigm in this project, so would like to avoid logic inside `catch()`.

TaskEither.orElse would help here, for example something like this piece of code above the TE.fold:

```ts
			TE.orElse((err) =>
				pipe(
					promiseToTE(
						() => AsyncStorage.getItem('LAST_KNOWN_LOCATION'),
						'LocationContext'
					),
					TE.chain((locationString) => {
						// Skip parsing if AsyncStorage is empty.
						if (!locationString) {
							return TE.left(err);
						}

						const parsedLocation = JSON.parse(
							locationString
						) as Location;

						// Make sure parsedLocation is well-formed.
						return parsedLocation.latitude &&
							parsedLocation.longitude
							? TE.right(parsedLocation)
							: TE.left(err);
					})
				)
			),
```

Reply:
thanks for the guidance, will take a look at this